### PR TITLE
PS-1855 optional metadata + runId

### PR DESCRIPTION
**Pull request by pivnicek**
I switched argument order to allow for default.

### CODE REVIEW

File: src/Writer/FileWriter.php [hunk 1]
```diff
@@ -17,12 +17,12 @@ class FileWriter extends AbstractWriter
...
      * @param array $configuration Upload configuration
-     * @param array $systemMetadata Metadata identifying the source of the file
      * @param string $storage Currently any storage that is not ABS workspaces defaults to local
+     * @param array $systemMetadata Metadata identifying the source of the file (defaults to empty array
      */
-    public function uploadFiles($source, $configuration, $systemMetadata, $storage)
+    public function uploadFiles($source, $configuration, $storage, $systemMetadata = [])
     {
-        if (empty($systemMetadata[self::SYSTEM_KEY_COMPONENT_ID])) {
+        if (!empty($systemMetadata) && empty($systemMetadata[self::SYSTEM_KEY_COMPONENT_ID])) {
```
Comment (odinuv) [inline src/Writer/FileWriter.php:25]: please leave it at is it was (you can still pass empty array upstream, which is ok)

**Review by odinuv — APPROVED** [review]

File: src/Writer/FileWriter.php (hunk 1 above)
Comment (pivnicek) [reply src/Writer/FileWriter.php:25]: I don't get it.  If I don't include the first condition, empty arrays will all produce this exception

Comment (odinuv) [reply src/Writer/FileWriter.php:25]: sorry, i missed a line, i was referring to the function signature

Comment (pivnicek) [reply src/Writer/FileWriter.php:25]: ah, ok.  I'll revert that.

**Comment by pivnicek** [comment]
sorry for the re-ping, just wanted to confirm

File: src/Writer/Helper/TagsHelper.php [hunk 2]
```diff
@@ -22,17 +22,19 @@ public static function rewriteTags(array $storageConfig, ClientWrapper $clientWr
...
-            if (in_array($systemKey, [
-                FileWriter::SYSTEM_KEY_COMPONENT_ID,
-                FileWriter::SYSTEM_KEY_CONFIGURATION_ID,
-                FileWriter::SYSTEM_KEY_CONFIGURATION_ROW_ID,
-                FileWriter::SYSTEM_KEY_BRANCH_ID,
-            ])) {
-                $storageConfig['tags'][] = $systemKey . ': ' . $systemValue;
-            } else {
-                $logger->info(sprintf('Not generating tag for key: %s', $systemKey));
+        if (!empty($systemMetadata)) {
```
Comment (odinuv) [inline src/Writer/Helper/TagsHelper.php:25]: unnecessary condition

**Review by odinuv — APPROVED** [review]

File: src/Writer/Helper/TagsHelper.php (hunk 2 above)
Comment (pivnicek) [reply src/Writer/Helper/TagsHelper.php:25]: 🤦 yeah

**Pull request merged**